quil-rs/src/program/scheduling/graph.rs

```diff
@@ -826,6 +843,32 @@ DECLARE integers INTEGER[1]
 LOAD bits[0] bits2 integers[0] # write
 LOAD bits[0] bits3 integers[0] # write
 LOAD bits4[0] bits integers[0] # read
+"#
+    }
+
+    build_dot_format_snapshot_test_case! {
+        memory_dependency_not_in_block_terminator,
+        r#"
+DECLARE ro BIT
+DECLARE depends_on_ro BIT
+
+NONBLOCKING CAPTURE 0 "ro_rx" flat(duration: 2.0000000000000003e-06, iq: 1.0, scale: 1.0, phase: 0.8745492960861506, detuning: 0.0) ro
+MOVE depends_on_ro ro
```

Comment:
What happens here if `depends_on_ro` is used in the _next_ block? Is that supported? I would make a test case for that, regardless as to whether it's supported. If not supported, a failing test case, issue, and some documentation on the limitation would be nice.

Reply:
We only track dependencies within blocks because there's already a total ordering constraint between blocks; we don't do inter-block scheduling.  (This is, AFAIK, already documented.)  So it's not supported but because it's not really relevant.  A test case is a fine idea, but I don't think it will fail and don't think there's an issue to create.